scripts/migrations/2024_10_remove_duplicate_products_on_wrong_flavors.pl

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2019 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use CGI::Carp qw(fatalsToBrowser);
+
+# use ProductOpener::PerlStandards;
+# not available in old versions of ProductOpener running on obf, opf, opff
+
+use ProductOpener::PerlStandards;
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Store qw/:all/;
+use ProductOpener::Tags qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Paths qw/:all/;
+use ProductOpener::Data qw/:all/;
+use ProductOpener::Orgs qw/:all/;
+use ProductOpener::Redis qw/:all/;
+
+use CGI qw/:cgi :form escapeHTML/;
+use Storable qw/dclone/;
+use Getopt::Long;
+use File::Copy (qw/move/);
+
+my $usage = <<TXT
+Usage:
+
+./2024_10_remove_duplicate_products_on_wrong_flavors.pl --flavor-code-csv [CSV file with flavor and code columns]
+
+This script reads a tab separated CSV file on STDIN
+The first column is either empty or contains a flavor: off, obf, opf, or opff
+The second column is a barcode
+Other columns may be included, they were generated by the 2024_10_detect_duplicate_products_in_different_flavors.pl script
+The ouput of the detect duplicate script was reviewed manually to put in the first column which flavor should be kept.
+Reviewed file: https://docs.google.com/spreadsheets/d/1-2WMvUC4J7iRYe3587mHJ1htIxPFyo7JLDLKHVmSum0/edit?gid=1565589772#gid=1565589772
+
+The script will move the products that are not kept to the other-flavors-codes directory (for both product and images)
+
+TXT
+	;
+
+my $csv_file;
+
+GetOptions("flavor-code-csv=s" => \$csv_file,);
+
+if (not defined $csv_file) {
+	print STDERR $usage;
+	exit;
+}
+
+open(my $log, ">>", "$data_root/logs/remove_duplicate_products_on_wrong_flavors.log");
+print $log "move_ean8_products_to_new_path.pl started at " . localtime() . "\n";
```

Comment:
It's a bit obvious you copy pasted :wink: 
```suggestion
print $log "remove_duplicate_products_on_wrong_flavors.pl started at " . localtime() . "\n";
```